app/Providers/UserServiceProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Providers;
+
+use Illuminate\Support\ServiceProvider;
+use App\Services\UserService;
+
+class UserServiceProvider extends ServiceProvider
+{
+    /**
+     * Register services.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->app->bind('App\Services\UserService', function ($app) {
+            return new UserService();
+        });
```

Comment:
Viết này cũng được nhưng chỉ cần khi em bind interface, còn class bình thường thì Laravel tự resolve được.

Ví dụ khi viết:
```php
class UserService implements UserServiceInterface

        $this->app->bind('App\Services\UserServiceInterface', function ($app) {
            return new UserService();
        });
```

https://laravel.com/docs/8.x/container#simple-bindings
> There is no need to bind classes into the container if they do not depend on any interfaces. The container does not need to be instructed on how to build these objects, since it can automatically resolve these objects using reflection.

Reply:
Dạ vâng ạ